core/src/main/java/pl/agh/edu/engine/time/command/RepeatingTimeCommand.java

```diff
@@ -55,13 +55,13 @@ protected RepeatingTimeCommand(Frequency frequency,
 	}
 
 	@Override
-	public void execute(Runnable postAction) {
+	public boolean execute() {
```

Comment:
troche mi się nie podoba to rozwiązanie bo funkcjonalność jaką miała mieć tylko RepeatingTimeCommand i która ją odróżniała od TimeCommand czyli odpalanie się jeszcze raz zostaje też "udostępniona" dla TimeCommada. Przy czym TimeCommand celowo z tego nie korzysta bo zawsze zwraca false'a. Widzę dwa rozwiązania: 

1. Wywalić klase RepeatingTImeCommand skoro odpalanie się jeszcze raz juz nie leży w jej odpowiedzialności i przenieść jej działanie do zwykłegoTimeCommanda (czyli jakieś dodatkowe pole w TimeCommand ktore będzie mówić czy Commenda ma być jeszcze raz odpalona.
2. Wywalić zwracanie booleana przez metode exeucute. TimeCommaExecutor sprawdza przy uzyciu instanceOf z jaką klasą TImeCOmmand ma doczynienia i TimeCommandy wywala a RepeatingTimeCommandy jeszcze raz wstawia. To w sumie jest uproszczoną wersją tego co jest teraz (bo rozróżniamy jaka jest klasa TimeCOmmand czy Repeating przez to czy execute zwracał true lub false

Reply:
3. Opcja z polem isSerializable do klasy Command - wtedy klasa RepeatingTimeCommand będzie się sama wstawaiała do odpowiedniej kolejki. Mogły by być dwie metody addCommand() i addNonSerializableCommand w których środku ustawiana by była flaga isSerializable dla danej komendy - tak zeby nie trzeba było  tego w konstruktorze przekazywać.

Reply:
wg mnie: 3 > 1 > 2 > wersja obecna

Reply:
Według mnie użycie flagi która mówi nam czy komenda jest serializowana czy nie w momencie kiedy tak naprawdę nie ma to znaczenia bo jest serializować tylko wtedy kiedy jest w dobrej kolejce a flaga informuje nas tylko o pseudo referencji do której kolejki mamy ja ewentualnie zwrócić jest mocno przesadzonym rozwiązaniem. Uzuce instance of to już w ogóle nie ma sensu. Więc wywalam generalizację

Reply:
to może by zrobić tylko jedną kolejkę i podczas serializacji odfiltrować te komendy których nie chcemy serializować (na podstawie flagi)

Reply:
Trochę nie siedzi mi rozwiązanie które zrobi nam bardziej czysty kod kosztem efektywności